### PR TITLE
Collection to creator, see all creators, show twitter image & handle, filter counts, update graph types

### PR DESCRIPTION
**Pull request by alchemistgo87**
ADDRESSES - #167 , #103, #100 

- Changed routes from /collections to /creators (Route: /creators/[CreatorAddress])
- Created new page to 'See all' creators (Route: /creators)
- Update filter button design
- Added nft counts to filters
- Updated creators preview design on home page
- Updated creator (previously collection) page design
- Updated NFT Card with twitter handles and twitter profile images
- Updated Nft Detail page with owners' and creators' twitter images/handles
- Updated graph types (including wallet, nft, twitter profile, nft counts) according to the new changes in the indexer

Screenshot 2022-04-20 at 4 37 07 PM

Screenshot 2022-04-20 at 4 38 16 PM

**SEE ALL CREATORS PAGE >**

Screenshot 2022-04-20 at 4 30 44 PM

Screenshot 2022-04-14 at 5 02 35 PM

**CHANGED FROM COLLECTION TO CREATOR PAGE**

Screenshot 2022-04-08 at 9 25 00 PM

Screenshot 2022-04-14 at 5 03 39 PM

Screenshot 2022-04-14 at 5 04 11 PM

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -338,17 +450,19 @@ const Home: NextPage<HomePageProps> = ({ marketplace }) => {
               className="absolute border-4 object-cover border-gray-900 bg-gray-900 rounded-full w-28 h-28 -top-32"
             />
             <h1>{marketplace.name}</h1>
-            <p className="mt-4 max-w-prose">{marketplace.description}</p>
+            <p className="mt-4 max-w-prose text-gray-300">
```
Comment (kespinola) [inline src/pages/index.tsx:453]: Need a max width on the p noticed it stretches all the way to the stats block when the description is longer.

File: src/pages/index.tsx [hunk 2]
```diff
@@ -643,9 +868,7 @@ const Home: NextPage<HomePageProps> = ({ marketplace }) => {
               itemRender={(nft) => {
                 return (
                   <Link to={`/nfts/${nft.address}`} key={nft.address}>
-                    <a>
```
Comment (kespinola) [inline src/pages/index.tsx:646]: ty.

File: src/pages/creators/index.tsx [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+  console.log('Creators', creatorsQuery.data)
+
+  const [hasMore, setHasMore] = useState(true)
+
+  const loading = creatorsQuery.loading
+
+  return (
+    <div className="flex flex-col items-center text-white bg-gray-900">
+      <Head>
+        <title>{marketplace.name + ' Creators'}</title>
```
Comment (kespinola) [inline src/pages/creators/index.tsx:156]: es6 notation please.

File: src/pages/creators/index.tsx [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+                </div>
+                <div className="bg-gray-800 h-6 w-24 block" />
+              </div>
+            </>
+          ) : (
+            creatorsQuery.data?.marketplace.creators.map((creator) => {
+              return (
+                <Link
+                  className="transition-transform hover:scale-[1.02]"
+                  key={creator.storeConfigAddress}
```
Comment (kespinola) [inline src/pages/creators/index.tsx:225]:
```suggestion
                  key={creator.creatorAddress}
```

storeConfigAddress is the same for all the creators.

File: src/pages/creators/index.tsx [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+                            className="h-28 object-cover rounded-md grow"
+                            src={nft.image}
+                            key={nft.address}
+                            alt={nft.name}
+                          />
+                        )
+                      })}
+                    </div>
+                    <div className="flex xl:hidden w-full flex-grid mb-2 gap-4 overflow-hidden ">
+                      {creator.preview.slice(0, 2).map((nft) => {
```
Comment (kespinola) [inline src/pages/creators/index.tsx:260]: Api always returns 3. can drop the slice.
```suggestion
                      {creator.preview.map((nft) => {
```

**Review by kespinola — CHANGES_REQUESTED** [review]
Left general CR but the main functional change that is need is to account for the profile being potentially null which is the case when the wallet isnt connected to bonfida.

However, we want to always show an avatar image so in the client.ts do a Field resolve for TwitterProfile profileImageUrl that checks the value returned by the api and if null returns our gradient (same blue one for the wallet portal). If this is done then the profile Image will always be available but the handle may not. That is what you check to decide where to display handle or wallet address.

File: src/pages/creators/index.tsx (hunk 5 above)
Comment (alchemistgo87) [reply src/pages/creators/index.tsx:260]: @kespinola This is to show only 2 nft preview in small screens (Design Requirement)

File: src/pages/index.tsx (hunk 1 above)
Comment (alchemistgo87) [reply src/pages/index.tsx:453]: Sure, providing the required right margin for this.

File: src/pages/nfts/[address].tsx [hunk 6]
```diff
@@ -627,37 +639,74 @@ const NftShow: NextPage<NftPageProps> = ({ marketplace, nft }) => {
...
+                <div className="mb-1 label">
+                  {loading ? (
+                    <div className="h-4 bg-gray-800 rounded w-14" />
+                  ) : (
+                    <span className="text-gray-300 text-sm">Created by</span>
+                  )}
+                </div>
+                <div className="flex ml-1.5">
+                  {loading ? (
+                    <div className="w-20 h-6 bg-gray-800 rounded -ml-1.5" />
```
[the remark this answers is not in the document]
Comment (alchemistgo87) [reply src/pages/nfts/[address].tsx:653]: No, that is already been taken care of by providing a positive ml-1.5 in the parent element.

File: src/components/NftCard/NftCard.tsx [hunk 7]
```diff
@@ -33,8 +33,42 @@ export const NftCard = ({ nft, marketplace }: NftCardProps) => {
...
+            >
+              <img
+                alt={nft.creators[0].profile.handle}
+                className="rounded-full h-5 w-5 object-cover user-avatar border-2 border-gray-900"
+                src={nft.creators[0].profile.profileImageUrl}
+              />
+
+              <div className="text-xs font-medium text-gray-300 ">Creator</div>
+            </Link>
+          )}
```
[the remark this answers is not in the document]
Comment (alchemistgo87) [reply src/components/NftCard/NftCard.tsx]: @kespinola Is this what we are supposed to add to client file:

```
TwitterProfile: {
        fields: {
          profileImageUrl: {
            read: (value) => {
              if (!value) {
                return '<DEFAULT_IMAGE_URL>'
              }
              return value
            },
          },
        },
      },
```

Secondly, In the 'DEFAULT_IMAGE_URL' should I add the too long user-avatar gradient url provided in global.css file?

**Review by kespinola — APPROVED** [review]

**Pull request merged**